### ISSUE TITLE
@manipulate broken

### ISSUE THREAD

**Issue by AshtonSBradley** [JuliaGizmos/Interact.jl#315]
`@manipulate` code appears to be broken:

```julia
@manipulate for i in 1:10
   i
end
``` 
throws
```julia
┌ Warning: newid(prefix) is deprecated. Scopes don't need this anymore. To get the scope's id, use scopeid(scope)
│   caller = ip:0x0
└ @ Core :-1

MethodError: no method matching Scope(::String; imports=Any["knockout"=>"/Users/abradley/.julia/packages/Knockout/JE2Yq/src/../assets/knockout.js", "knockout_punches"=>"/Users/abradley/.julia/packages/Knockout/JE2Yq/src/../assets/knockout_punches.js"])
Closest candidates are:
  Scope(::Any, !Matched::Dict{String,Tuple{Observables.AbstractObservable,Union{Nothing, Bool}}}, !Matched::Set{String}, !Matched::Any, !Matched::Array{Asset,1}, !Matched::Any, !Matched::WebIO.ConnectionPool, !Matched::Array{WebIO.JSString,1}) at /Users/abradley/.julia/packages/WebIO/jo4DM/src/scope.jl:51 got unsupported keyword argument "imports"
  Scope(; id, tagstr, props, tag, dom, outbox, observs, private_obs, systemjs_options, imports, jshandlers, mount_callbacks) at /Users/abradley/.julia/packages/WebIO/jo4DM/src/scope.jl:121

Stacktrace:
 [1] #knockout#1(::Array{Pair{String,WebIO.JSString},1}, ::Array{Any,1}, ::Function, ::Node{WebIO.DOM}, ::Array{Pair{String,Any},1}, ::WebIO.JSString) at /Users/abradley/.julia/packages/Knockout/JE2Yq/src/Knockout.jl:28
 [2] (::getfield(Knockout, Symbol("#kw##knockout")))(::NamedTuple{(:computed,),Tuple{Array{Pair{String,WebIO.JSString},1}}}, ::typeof(knockout), ::Node{WebIO.DOM}, ::Array{Pair{String,Any},1}, ::WebIO.JSString) at ./none:0
 [3] #input#47(::WebIO.JSString, ::Array{Pair{String,Array{String,1}},1}, ::Nothing, ::String, ::String, ::String, ::Dict{Any,Any}, ::Bool, ::Array{Pair{String,WebIO.JSString},1}, ::Dict{Any,Any}, ::String, ::String, ::String, ::Int64, ::Base.Iterators.Pairs{Symbol,Int64,Tuple{Symbol,Symbol,Symbol},NamedTuple{(:min, :max, :step),Tuple{Int64,Int64,Int64}}}, ::typeof(input), ::Interact.Bulma, ::Int64) at /Users/abradley/.julia/packages/InteractBase/l8cEC/src/input.jl:258
 [4] (::getfield(Widgets, Symbol("#kw##input")))(::NamedTuple{(:bindto, :attributes, :extra_obs, :computed, :typ, :min, :max, :step, :className),Tuple{String,Dict{Any,Any},Array{Pair{String,Array{String,1}},1},Array{Pair{String,WebIO.JSString},1},String,Int64,Int64,Int64,String}}, ::typeof(input), ::Interact.Bulma, ::Int64) at ./none:0
 [5] #input#40(::Base.Iterators.Pairs{Symbol,Any,NTuple{9,Symbol},NamedTuple{(:bindto, :attributes, :extra_obs, :computed, :typ, :min, :max, :step, :className),Tuple{String,Dict{Any,Any},Array{Pair{String,Array{String,1}},1},Array{Pair{String,WebIO.JSString},1},String,Int64,Int64,Int64,String}}}, ::Function, ::Int64) at /Users/abradley/.julia/packages/Widgets/AcUe0/src/defaults.jl:10
 [6] (::getfield(Widgets, Symbol("#kw##input")))(::NamedTuple{(:bindto, :attributes, :extra_obs, :computed, :typ, :min, :max, :step, :className),Tuple{String,Dict{Any,Any},Array{Pair{String,Array{String,1}},1},Array{Pair{String,WebIO.JSString},1},String,Int64,Int64,Int64,String}}, ::typeof(input), ::Int64) at ./none:0
 [7] #slider#81(::String, ::Bool, ::String, ::Int64, ::String, ::Dict{Any,Any}, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Interact.Bulma, ::UnitRange{Int64}, ::Array{String,1}) at /Users/abradley/.julia/packages/InteractBase/l8cEC/src/slider.jl:58
 [8] (::getfield(Widgets, Symbol("#kw##slider")))(::NamedTuple{(:label,),Tuple{String}}, ::typeof(slider), ::Interact.Bulma, ::UnitRange{Int64}, ::Array{String,1}) at ./none:0
 [9] (::getfield(Widgets, Symbol("#kw##slider")))(::NamedTuple{(:label,),Tuple{String}}, ::typeof(slider), ::Interact.Bulma, ::UnitRange{Int64}) at ./simdloop.jl:0
 [10] #slider#56(::Base.Iterators.Pairs{Symbol,String,Tuple{Symbol},NamedTuple{(:label,),Tuple{String}}}, ::Function, ::UnitRange{Int64}) at /Users/abradley/.julia/packages/Widgets/AcUe0/src/defaults.jl:10
 [11] (::getfield(Widgets, Symbol("#kw##slider")))(::NamedTuple{(:label,),Tuple{String}}, ::typeof(slider), ::UnitRange{Int64}) at ./none:0
 [12] #widget#104(::Base.Iterators.Pairs{Symbol,String,Tuple{Symbol},NamedTuple{(:label,),Tuple{String}}}, ::Function, ::UnitRange{Int64}) at /Users/abradley/.julia/packages/Widgets/AcUe0/src/defaults.jl:56
 [13] (::getfield(Widgets, Symbol("#kw##widget")))(::NamedTuple{(:label,),Tuple{String}}, ::typeof(widget), ::UnitRange{Int64}) at ./none:0
 [14] top-level scope at /Users/abradley/.julia/packages/Widgets/AcUe0/src/manipulate.jl:91
 [15] top-level scope at In[7]:1
```
with versions:
```julia
Julia Version 1.1.0
Commit 80516ca202 (2019-01-21 21:24 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 8
```

**Comment by AshtonSBradley** [comment]
Is there a recommended earlier version I can pin this to that works for julia 1.0.2? I have a computational physics exam I am running very soon and I have to establish a stable package environment before web access gets killed...

**Comment by travigd** [comment]
I think a constructor must have been borked™ when https://github.com/JuliaGizmos/WebIO.jl/pull/295 was merged. This constructor was technically deprecated for a while but because of reasons (something something dispatch precedence something something) the method that was marked as deprecated was never being called.

I'll fix this asap and release 0.8.4.

**Comment by AshtonSBradley** [comment]
that's strange -`Interact.jl` pass. Maybe worth adding this `@manipuate` test?

**Comment by travigd** [comment]
It's because WebIO is a dependency of Interact and dependency updates don't trigger rebuilds of upstream projects. It's basically WebIO's fault for not having sufficient test cases to detect that we had a breaking change (we'll never be 100% perfect but we could definitely be better).

**Comment by AshtonSBradley** [comment]
The error is fixed, but now the plot doesn't update with the slider:
```julia
using Plots, Interact
x = LinRange(0,10,100)
@manipulate for i in 1:10
    plot(x,cos.(i*x))
end
```
produces (for any slider value)

<img width="971" alt="Screen Shot 2019-06-07 at 3 54 40 PM" src="https://user-images.githubusercontent.com/24194258/59080391-973e5b00-893c-11e9-8c1f-754e9f18415c.png">

**Comment by AshtonSBradley** [comment]
or, even simpler:
```julia
@manipulate for i in 1:10
    @show i
end
```

<img width="1025" alt="Screen Shot 2019-06-07 at 3 56 08 PM" src="https://user-images.githubusercontent.com/24194258/59080442-d5d41580-893c-11e9-9e52-c5c6cad92cc4.png">

**Comment by AshtonSBradley** [comment]
@travigd I can report it is also broken in Juno

**Comment by travigd** [comment]
For posterity, here's the rough sequence of events that caused this to break.

* We deprecated scopes having manually-set ids (instead just using `objectid`).
* We forgot to patch the `Scope(id; kwargs...)` method. The method had been deprecated for a while, but because of the way things worked™, the deprecation warnings were getting swallowed. (I think this had to do with there being two methods that matched that signature, only one of which was deprecated properly).
* We fixed that but fixed it wrong. It was essentially `Scope(id; kwargs...) = Scope(kwargs...)` - which meant that all of the `Pair`s of keyword arguments were being handled as positional arguments.
This effectively meant that `Scope(id; imports=["foo.js"])` was mapped to `Scope(:imports => "foo.js")` (as a positional argument) which then was recursively mapped to `Scope()`.

This broke the upstream package Knockout.jl ([I also fixed the usage of deprecated methods](https://github.com/JuliaGizmos/Knockout.jl/pull/16)) and resulted in an error when Knockout tried to setup it's JavaScript (since the imports it passed in weren't getting... y'know... imported).

I added a deprecation test to WebIO for this specific case, but it might be worth it to add some Interact specific stuff to our test suite for WebIO to ensure we're not breaking downstream things. This gets kind of messy when we do breaking changes (which I'm about 10000% certain we will if/when we get around to releasing WebIO 1.0) so I'm not totally sure. I'd also like to just up the coverage on WebIO in general and hopefully avoid this kind of thing from recurring.

**Comment by jleman** [comment]
I observed the same behavior in Jupyter after updating Julia to 1.1.1 and updating all packages  (Windows 10 machine).  I'm using Interact v0.10.2.  I found the following process seems to fix the issue temporarily, but I have to repeat it every time I open a notebook containing and Interact script.

1. First, with the notebook open, go to "Kernel"->"Change Kernel" and select an older Julia Kernel (in my case 1.0.3). Wait for connection to the kernel to establish.
2. Second, using the same menu items, change back to the newest kernel (1.1.1 in my case)

For me this causes the Interact functionality to work, though it is tedious.  Hopefully this provides a clue as to what's wrong.

Here is the code block I am running:
```
using Interact
@manipulate for t=1:NDT
plot(collect(1:NDZ),Vout[t,:])
end
```

The script links a slider to plot rows of matrix Vout.
